python/ray/workflow/tests/test_wait_for_event.py

```diff
@@ -0,0 +1,26 @@
+import ray
+from ray import workflow
+from ray.workflow.event_listener import EventListener
+
+ray.init(address='auto')
+workflow.init()
+
+class ExampleEventProvider(EventListener):
+    def __init__(self):
+        pass 
+
+    async def poll_for_event(self, *args):
+        pass
+
+    async def event_checkpointed(self, *args):
+        pass
+
+
+@workflow.step
+def handle_event(msg):
+    print(msg)
+
+event_func = workflow.wait_for_event_revised(ExampleEventProvider, "hello")
```

Comment:
@yuanchi2807  do you mean here the event step is not executed right?

The reason is that it's not handled in step_executor. All step type need to be handled in step executor otherwise, it won't know how to execute this step.

Reply:
Turned out the answer is by getting workflow.wait_for_event to return WorkflowData.func_body=lamdba:None. This object is not meant to do anything but to carry step_options StepType.EVENT

step_options = WorkflowStepRuntimeOptions.make(
        step_type=StepType.EVENT
    )
    workflow_data = WorkflowData(
        func_body=lambda:None,
        inputs=event_inputs,
        step_options=step_options,
        name=None, #f"{current_step_id}.event",
        user_metadata={},
    )
    logger.info(f"wait_for_event_revised exit")
    return Workflow(workflow_data)

Reply:
@iycheng If I remember correctly, you seemed to have suggested that it is in `_workflow_step_executor `where the control is passed to the `EventCoordinator.`  If the func_body of the `workflow.wait_for_event()` is also talking to the `EventCoordinator`, then we will end up with two different places that the` EventCoordinator` need to communicate with.